tesseract_core/sdk/engine.py

```diff
@@ -268,6 +277,7 @@ def build_image(
     image_name: str,
     dockerfile: str | Path,
     build_dir: str | Path,
+    requirements: PythonRequirements,
```

Comment:
It reads cleaner to me to pass the entire `user_config` here (same as for `create_dockerfile`).